src/lib/OS/SimOS.cc

```diff
@@ -288,6 +291,10 @@ void SimOS::terminateThread(uint64_t tid) {
   // tick()
   proc->second->status_ = procStatus::completed;
   // Remove from processes_
+  // There is no need to remove the FutexInfo struct associated with this
+  // process. Since this process is in procStatus::executing state it won't have
+  // an associated FutexInfo struct. When a process is woken up via the
+  // FUTEX_WAKE flag the FutexInfo struct associated with it is destroyed.
```

Comment:
A sleeping thread could be terminated. What would happen if  FUTEX_WAKE is performed on a prevously terminated thread? Could be re-scheduled errornously without a reference in the main processes_ structure

Reply:
fixed